cmd/clusterctl/phases/pivot.go

```diff
@@ -219,24 +276,30 @@ func moveMachineDeployments(from sourceClient, to targetClient, machineDeploymen
 
 	for _, md := range machineDeployments {
 		if err := moveMachineDeployment(from, to, md); err != nil {
-			return errors.Wrapf(err, "failed to move MachineDeployment %s:%s", md.Namespace, md.Name)
+			return errors.Wrapf(err, "failed to move MachineDeployment %s/%s", md.Namespace, md.Name)
 		}
 	}
 	return nil
 }
 
 func moveMachineDeployment(from sourceClient, to targetClient, md *clusterv1.MachineDeployment) error {
 	klog.V(4).Infof("Moving MachineDeployment %s/%s", md.Namespace, md.Name)
+
 	klog.V(4).Infof("Retrieving list of MachineSets for MachineDeployment %s/%s", md.Namespace, md.Name)
 	machineSets, err := from.GetMachineSetsForMachineDeployment(md)
 	if err != nil {
 		return err
 	}
-
 	if err := moveMachineSets(from, to, machineSets); err != nil {
```

Comment:
When we discussed this previously, we said we could probably simplify the logic quite a bit by making sure the controllers were not running in both the source and target clusters during the pivot, then we wouldn't have to worry about ordering of resources and could just migrate wholesale.

Reply:
We definitely can, it seems like a good follow-up improvement task. The trick there is to scale down the target cluster deployments and then scale them back up at the end.

Reply:
Or to filter the provider components to not apply the Deployments prior to migrating the resources, and only apply them afterwards.

Reply:
We also talked about having pivot (or future clusteradm) add either an annotation or a label to each object that's being pivoted in the source cluster, then adding code to all our controllers to no-op and return ASAP if they see that label/annotation. That way, we don't have to worry about the controllers racing.